src/test/java/org/opensearch/knn/plugin/action/RestGetModelHandlerIT.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.plugin.action;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.opensearch.action.ActionFuture;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.knn.KNNSingleNodeTestCase;
+import org.opensearch.knn.index.SpaceType;
+import org.opensearch.knn.index.util.KNNEngine;
+import org.opensearch.knn.indices.Model;
+import org.opensearch.knn.indices.ModelDao;
+import org.opensearch.knn.indices.ModelMetadata;
+import org.opensearch.knn.indices.ModelState;
+import org.opensearch.knn.plugin.transport.GetModelAction;
+import org.opensearch.knn.plugin.transport.GetModelRequest;
+import org.opensearch.knn.plugin.transport.GetModelResponse;
+
+import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import static org.opensearch.knn.common.KNNConstants.MODEL_INDEX_NAME;
+
+/**
+ * Integration tests to check the correctness of {@link org.opensearch.knn.plugin.rest.RestGetModelHandler}
+ */
+
+public class RestGetModelHandlerIT extends KNNSingleNodeTestCase {
```

Comment:
Because Integration tests are used by infra to test remote clusters, this cannot be a SingleNodeTestCase. It needs to be a rest test case.

Reply:
Ack